v2/server.go

```diff
@@ -491,7 +508,9 @@ func (s *Server) Start() {
 		err := s.httpServer.ListenAndServe()
 		if err != nil {
 			log.Errorf("restarting due to error with api server %s\n", err.Error())
+			s.statusLock.Lock()
```

Comment:
s.SetStatus(failed)

Reply:
also original line 485:
s.SetStatus(started)

Reply:
fixed it ,also added safe close pattern for channels to pass race test